src/tqdm_publisher/_demos/_parallel_bars/_server.py

```diff
@@ -107,43 +111,45 @@ def _run_sleep_tasks_in_subprocess(
         time.sleep(sleep_time)
 
 
-def run_parallel_processes(request_id, url: str):
+def run_parallel_processes(*, all_task_times: List[List[float]], request_id: str, url: str):
 
     futures = list()
     with ProcessPoolExecutor(max_workers=N_JOBS) as executor:
 
         # # Assign the parallel jobs
-        for iteration_index, task_times in enumerate(TASK_TIMES):
+        for iteration_index, task_times_per_job in enumerate(all_task_times):
             futures.append(
                 executor.submit(
                     _run_sleep_tasks_in_subprocess,
-                    task_times=task_times,
+                    task_times=task_times_per_job,
                     iteration_index=iteration_index,
                     request_id=request_id,
                     url=url,
                 )
             )
 
         total_tasks_iterable = as_completed(futures)
-        total_tasks_progress_bar = TQDMPublisher(
+        total_tasks_progress_bar = TQDMProgressPublisher(
             iterable=total_tasks_iterable, total=len(TASK_TIMES), desc="Total tasks completed"
         )
 
+        # The 'total' progress bar bas an ID equivalent to the request ID
         total_tasks_progress_bar.subscribe(
             lambda format_dict: forward_to_http_server(
                 url=url, request_id=request_id, progress_bar_id=request_id, **format_dict
             )
         )
 
+        # Trigger the deployment of the parallel jobs
         for _ in total_tasks_progress_bar:
             pass
 
 
-def format_sse(data: str, event=None) -> str:
-    msg = f"data: {json.dumps(data)}\n\n"
+def format_sse(data: str, event: Union[str, None] = None) -> str:
```

Comment:
Can you explain the difference between the `data` and `event`?

Reply:
Server-sent events allow you to subscribe to named events, otherwise the default is `message`. 

Here is some example code from the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) on the `EventSource` class in JavaScript: 
```
const sse = new EventSource("/api/v1/sse");

/*
 * This will listen only for events
 * similar to the following:
 *
 * event: notice
 * data: useful data
 * id: someid
 */
sse.addEventListener("notice", (e) => {
  console.log(e.data);
});

/*
 * Similarly, this will listen for events
 * with the field `event: update`
 */
sse.addEventListener("update", (e) => {
  console.log(e.data);
});

/*
 * The event "message" is a special case, as it
 * will capture events without an event field
 * as well as events that have the specific type
 * `event: message` It will not trigger on any
 * other event type.
 */
sse.addEventListener("message", (e) => {
  console.log(e.data);
});
```

Reply:
What would you think about always having an explicit `event` header type, even in that last case

Such as

```python
def format_server_side_event(data: str, event_type: Literal["notice", "update", "message"] = "message")
    ...
    event_data = f"event: {event_type}\ndata:{data}"
```

?

Reply:
That's fine. This behavior is simply something we don't want to block for users—though we only really need to use "message" for our demos.

Reply:
Oh, there are more (arbitrary) types allowed for any event listener? I guess that's my confusion

Reply:
Yes the event name is entirely arbitrary, aside from the fact that failing to provide an event type defaults to "message" on the client-side.